### ISSUE TITLE
Option for focusing terminal

### ISSUE THREAD

**Issue by udondan** [jscheffner/vscode-workspace-terminals#4]
This is an awesome extension!

The only thing I can complain about is that the terminal is automatically opened when selecting a file from a workspace. Can it be switched in the background? I'd just like the correct terminal for the workspace if I open the terminal myself. If that's possible it would be great if there was setting to change this.

Cheers,
Daniel

**Comment by udondan** [comment]
I think it actually does not need an option. If the terminal is not visible, it should not open it. I haven't found a good solution though. I cannot see a method in the API that allow to check if the terminal view is visible at all. For now I hacked it locally by adding a `hide()` right after the `show()`. Of course that hides it if it's visible... :-/

Since it might not be clear, why this is an issue for me, I'll explain it a little better. I use my [terminals in full screen](https://marketplace.visualstudio.com/items?itemName=samueltscott.maximizeterminal). When switching to a different workspace in the file tree, it always opens the terminal in almost-full-screen:

![Bildschirmfoto 2020-07-15 um 12 18 35](https://user-images.githubusercontent.com/6443408/87533863-86126280-c695-11ea-91e0-f2ed363ef5ff.png)

**Comment by jscheffner** [comment]
Hey, thanks for the feedback. That is a known issue (see #3). Unfortunately, the VS Code extension API does not provide any way to find out if the terminal panel is open.

**Comment by udondan** [comment]
Ah, didn't see that one as it was closed, sorry.

> Unfortunately, the VS Code extension API does not provide any way to find out if the terminal panel is open.

Yeah, that's what I had fond out too. Too bad :-/

**Comment by bhipple** [comment]
That's unfortunate.  I really want my terminal to swap to the workspace root I'm in when I open it, but I don't want to permanently keep the terminal open and waste the screen space when I don't need it.

I'm relatively new to VSCode and not sure what the options are, but is there a way I can bind `CTRL ~` to `getActiveWorkspaceTerminal`, so that when I press the key to open a terminal it opens the one this plugin would?

**Comment by jscheffner** [comment]
@bhipple , if you open the Command Palette (⇧⌘P) and search for the "Open Workspace Terminal" command, you can see a gear icon. If you click it, you should be able to configure a keybinding.

**Comment by jscheffner** [comment]
<img width="719" alt="Screenshot 2022-09-18 at 21 30 01" src="https://user-images.githubusercontent.com/14218270/190925336-c57cdd2c-c78a-4af3-b694-13582f955f54.png">

**Comment by bhipple** [comment]
Hmm, when I call that function it doesn't seem to do anything.  If I call `View: Toggle Terminal` (default vscode function), it pops up  at the bottom, and I can see all my workspace terminals are running.  But it goes to whichever was last active, not the one for my buffer's workspace; and calling `Open Workspace Terminals` doesn't seem to change it.  Does this work for you?

**Comment by jscheffner** [comment]
You're right, the command only opens the terminal if it just created it, not if it already existed before. I will have a look at it. Probably a second command for switching to the right terminal would be the best solution.

**Comment by ssbarnea** [comment]
I am closely looking for the development of this as I really wish we could end-up with having a single key to show/hide terminal in vscode, one that knows to open a new terminal if needed.

I found the ability to auto-hide/show the left sidebar quite useful, so maybe we can end-up with something similar, so we can maximize the size of edit window and still be able quickly check the terminal. Sadly vscode will likely never get real multi-monitor support, so we need to make the switching as easy as possible.

**Comment by sebastiantf** [comment]
Would love for this to happen somehow. Sad that vscode api doesnt help
